profiling/pthread_everywhere.h

```diff
@@ -75,6 +75,9 @@ inline void pthread_cond_signal(pthread_cond_t* cond) {
 inline void pthread_cond_wait(pthread_cond_t *cond, pthread_mutex_t *mutex) {
   std::unique_lock<std::mutex> lock(**mutex, std::adopt_lock);
   (*cond)->wait(lock);
+  // detach lock from mutex so when we leave this conext
```

Comment:
good catch. i'm confused that i didn't catch that in my local testing.

Reply:
I think only test_blocking_counter.cc was running into this

Reply:
i see, thanks.